server/etcdserver/apply/apply_auth.go

```diff
@@ -125,6 +125,11 @@ func (aa *authApplierV3) LeaseRevoke(lc *pb.LeaseRevokeRequest) (*pb.LeaseRevoke
 func (aa *authApplierV3) checkLeasePuts(leaseID lease.LeaseID) error {
 	l := aa.lessor.Lookup(leaseID)
 	if l != nil {
+		// early return for most-common scenario of either disabled auth or admin user
+		if aa.as.IsAdminPermitted(&aa.authInfo) == nil {
+			return nil
```

Comment:
Do we have a test covering this case?

Reply:
Neither apply.go, nor apply_auth.go have unit test coverage-  there are no unit tests as far as I can tell.

Reply:
My suggestion would be to add `ArePutsPermitted` method that will for the range and add tests for it.